controllers/s3bucket_controller.go

```diff
@@ -128,3 +130,15 @@ func (r *S3BucketReconciler) handleDeleteFlow(bucketSpec *s3operatorv1.S3BucketS
 	isDelted, err := r.AwsClient.HandleBucketDeletion(bucketName)
 	return isDelted, err
 }
+func (r *S3BucketReconciler)updateStatus(err error, s3Bucket *s3operatorv1.S3Bucket){
```

Comment:
the method signature will minimize our ability in the future to expend the status and control it. i think a better way will be to give the bucket status as an argument and the method will check update the k8s resource (and not decide what is the status). in addition i think 'updateBucketResourceStatus' will be a better name